hudi-client/src/main/java/org/apache/hudi/index/bloom/IntervalTreeBasedGlobalIndexFileFilter.java

```diff
@@ -58,10 +67,12 @@
   }
 
   @Override
-  public Set<String> getMatchingFiles(String partitionPath, String recordKey) {
-    Set<String> toReturn = new HashSet<>();
-    toReturn.addAll(indexLookUpTree.getMatchingIndexFiles(recordKey));
-    toReturn.addAll(filesWithNoRanges);
+  public Set<Pair<String, String>> getMatchingFilesAndPartition(String partitionPath, String recordKey) {
+    Set<String> matchingFiles = new HashSet<>();
```

Comment:
can we try to use java8 collect() APIs to implement these.

Reply:
Are you suggesting something like this

Stream<String> matchingFiles = Stream.concat(filesWithNoRanges.stream(), indexLookUpTree.getMatchingIndexFiles(recordKey).stream());
    Set<Pair<String, String>> toReturn = new HashSet<>();
    matchingFiles.forEach(file -> toReturn.add(Pair.of(fileToPartitionPathMap.get(file), file)));
    return toReturn;

Reply:
Instead of forEach we can do a `collect(Collectors.toSet)` .. It's fine to leave it as-is. just a cosmetic suggestion